airflow-core/tests/unit/models/test_taskinstance.py

```diff
@@ -683,6 +683,32 @@ def test_next_retry_datetime_short_or_zero_intervals(self, dag_maker, seconds):
         date = ti.next_retry_datetime()
         assert date == ti.end_date + datetime.timedelta(seconds=1)
 
+    def test_next_retry_datetime_with_custom_multiplier(self, dag_maker):
+        delay = datetime.timedelta(minutes=4)
+
+        with dag_maker(dag_id="fail_dag"):
+            task = BashOperator(
+                task_id="task_with_custom_multiplier",
+                bash_command="exit 1",
+                retries=3,
+                retry_delay=delay,
+                retry_exponential_backoff=True,
+                retry_delay_multiplier=5.0,
+            )
```

Comment:
Yeah, I think this would be nicer as

```suggestion
            task = BashOperator(
                task_id="task_with_custom_multiplier",
                bash_command="exit 1",
                retries=3,
                retry_delay=delay,
                retry_exponential_backoff=5,
            )
```

(i.e. needing to specify two variables feels clunky)

Reply:
looks good! In this case, is backward compatibility with previous version Dags okay?

Reply:
I like the idea, i've went ahead and replaced the boolean with a float and removed the new parameter. I've added backwards compatibility where True = 2.0 / False = 0.0